app/models/manageiq/providers/infra_manager.rb

```diff
@@ -44,5 +44,22 @@ def self.merge_config_settings(cfg = VMDB::Config.new("vmdb"))
       path = [:ems, :ems_redhat, :service, :read_timeout]
       cfg.merge_from_template_if_missing(*path)
     end
+
+    def self.queue_credential_validation(options)
```

Comment:
The typical pattern for queuing is to have the method being executed, say, in this case, `credential_validation`, and then a separate queueing method named `queue_credential_validation`.  This latter method would hold have the method you currently have in the controller (lines 451-466 or so).

Not sure I like the method name `credential_validation` though.  We already have `validate_credentials`, `authentication_check` and name-wise they all feel the same.  We need to find a way to untangle this mess :frowning: